### PR TITLE
Use Valgrind to detect unitialized variable references.

### PR DESCRIPTION
**Pull request by racer-coder**
Fix the edge cases that pop up.

### CODE REVIEW

File: .github/workflows/build-unit-tests.yaml [hunk 1]
```diff
@@ -70,3 +64,14 @@ jobs:
     - name: Run Tests (sharded)
       working-directory: ./unit_tests/
       run: bash ./run_sharded_tests.sh
+
+    - name: Rebuild Tests For Valgrind
```
Comment (rusefillc) [inline .github/workflows/build-unit-tests.yaml:68]: please add two extra lines explaining why rebuild

**Comment by rusefillc** [comment]
please provide some ELI5 documentation on the forum or wiki

how are errors showing themselves? what kind of errors is this toolset guarding against?

**Comment by racer-coder** [comment]
Google says ELI5 is a Python library on machine learning?

Can I edit the wiki?

**Comment by rusefillc** [comment]
https://www.reddit.com/r/explainlikeimfive/

yes
